lightshow/postprocess/autobroaden.py

```diff
@@ -0,0 +1,289 @@
+from scipy.stats import norm
+from scipy.stats import cauchy
+from scipy.special import voigt_profile as voigt
+
+import numpy as np
+
+
+class Site_spectum:
+    def __init__(
+        self,
+        xin,
+        yin,
+        VASP_out = [float, float, float],
+        weight = 1.0,
+        
+    ):
+        """
+        The spectrum information of a each site
+        Args:
+            xin, yin: simulated spectrum for each site
+            VASP_out: total energy of the final state with core hole, fermi energy of the final state, 
+            and total energy of the initial state without core hole
+            weight: the site weight in the crystal 
+        """
+        
+        self.xin = xin
+        self.yin = yin
+        self.vasp_out = VASP_out
+        self.Weight = weight
+        
+    @property
+    def spectrum(self):
+        return self.xin, self.yin
+    
+    @property
+    def weight(self):
+        return self.Weight
+    
+    @property
+    def VASP_out(self):
+        return self.vasp_out
+
+    
+class Broaden:
+    def __init__(
+        self,
+        core_state = float,
+        sigma = 1.0,
+        gamma = 1.0,
+        EXPshift = None
+    ):
+        """
+        Args:
+            core_state: Core state, e.g., 1s state energy for K-edge
+            EXPshift: Empirical constant shift to match theory to experiment
+        """
+      
+        self.core_state = core_state
+        self.sigma = sigma
+        self.gamma = gamma
+        self.EXPshift = EXPshift
+
+    
+    @staticmethod
+    def gauss_broaden(x,xin,yin,sigma):
+        yout = np.zeros_like(x)
+        
+        num=len(xin)
+        for i in range(num):
+
+            yout=yout+yin[i]*norm.pdf(x, xin[i], sigma)/num   
+        return yout 
+    
+    @staticmethod
+    def lorentz_broaden(x,xin,yin,gamma):
+        yout = np.zeros_like(x)
+        
+        num=len(xin)
+        for i in range(num):
+
+            yout=yout+yin[i]*cauchy.pdf(x, xin[i], gamma/2)/num   
+        return yout 
+   
+
+    @staticmethod
+    def voigt_broaden(x,xin,yin,sigma,gamma):
+        yout = np.zeros_like(x)
+        
+        num=len(xin)
+        for i in range(num):
+
+            yout=yout+yin[i]*voigt(x-xin[i],sigma,gamma/2)/num   
+        return yout 
+    
+
+    def crystal_spectrum_broaden(self, x, site_spectra, broad_method , sigma = None, gamma = None , shift = 0, cross_sec = True):
+        
+        yout = np.zeros_like(x)
+        
+        weight = 0
+            
+        for site_spectrum in site_spectra:
+            xin, yin = site_spectrum.spectrum
+        
+            if self.EXPshift != None:
+                xin = xin + self.EXPshift+site_spectrum.VASP_out[0]-site_spectrum.VASP_out[1]-site_spectrum.VASP_out[2]+self.core_state
+            else:   
+                xin = xin + shift
+                
+            if cross_sec:
+                yin = yin * xin
+                
+            if broad_method == "gauss":
```

Comment:
Recommend a different method for each of these instead of passing an extra argument. E.g.

```python
def gaussian_broaden(...):
    ...

def lorentzian_broaden(...):
    ...

def voigt_broaden(...):
    ...
```